students/Chao/session04/trigram.py

```diff
@@ -1,5 +1,71 @@
 #!/usr/bin/env python
 
+import string
+import random
+
 # Create an empty dictionary
 trig = {}
+# An empty list for word pool
+wpool = []
+
```

Comment:
These variables don't need to be here. It would be better if they were initiated in the functions that use them, and then passed as input/output variables. Really, the only variables that should be outside of the functions are constants, iow variables that always have the same value.